tock/hours/float.py

```diff
@@ -0,0 +1,147 @@
+            float_people_id=float_people_id,
+            weeks=26
+        )
+        work_days = get_work_days(
+            holidays=float_holidays,
+            start_date=rp.start_date,
+            end_date=rp.end_date
+        )
+        work_hours = get_work_hours(
+            start_date=rp.start_date,
+            end_date=rp.end_date,
+            timeoffs=float_timeoffs,
+            work_days=work_days
+        )
+        return get_tasks(
+            start_date=rp.start_date,
+            end_date=rp.end_date,
+            tasks=float_tasks,
+            work_days=work_days,
+            work_hours=work_hours
+        )
+    return None
+
+def get_tasks(start_date, end_date, tasks, work_days, work_hours):
+    for i in tasks:
+        t_start_date, t_end_date = \
+            [ dt.datetime.strptime(i[k], '%Y-%m-%d').date() for \
+            k in ['start_date', 'end_date'] ]
+        task_days = len(find_common_weekdays(
+            (start_date, end_date),
+            (t_start_date, t_end_date)
+        ))
+        i['hours_wk'] = float(i['hours_pd']) * task_days
+    all_task_hours = sum([ float(i['hours_wk']) for i in tasks ])
+    tasks_to_hours = all_task_hours / work_hours
+    if tasks_to_hours > 1.0:
+        for i in tasks:
+            week_hours = i['hours_wk'] * (work_hours / all_task_hours)
+            i.update({'hours_wk': round(week_hours, 2)})
+    return tasks
+
+def get_work_hours(start_date, end_date, timeoffs, work_days):
+    hours_off = 0
+    for i in timeoffs:
+        to_start_date, to_end_date = \
+            [ dt.datetime.strptime(i[k], '%Y-%m-%d').date() for \
+            k in ['start_date', 'end_date'] ]
+        days_off = len(find_common_weekdays(
+            (start_date, end_date),
+            (to_start_date, to_end_date)
+        ))
+        hours_off += float(i['hours']) * days_off
+    return (work_days * BASE_HOURS_PER_DAY) - hours_off
+
+def get_work_days(holidays, start_date, end_date):
+    # Get holiday info from Float for last two weeks
+    # and check for holidays in period.
+    return BASE_DAYS_PER_WEEK - len([ i for i in holidays \
+        if start_date <= \
+        dt.datetime.strptime(i['date'], '%Y-%m-%d').date() <= \
+        end_date ])
+
+def get_float_timeoffs(end_date, float_people_id, weeks=26):
+    # Get Float timeoff for the user data.
+    # Defaults to 26 weeks to guard against a really, really long
+    # absence.
+    return [ i for i in get_float_data(
+        endpoint='timeoffs',
+        params={
+            'weeks': weeks,
+            'start_day': end_date - dt.timedelta(weeks=weeks)
+        }
+    ).json()['timeoffs'] if int(i['people_id']) == float_people_id ]
+
+def get_float_holidays(start_date, weeks=2):
+    return get_float_data(
+        endpoint='holidays',
+        params={'start_day': (start_date - dt.timedelta(
+            weeks=weeks))
+        }
+    ).json()['holidays']
+
+def get_float_people_id(userdata):
```

Comment:
what do you have against blank lines? :)

not changing any of the code, i would, in general, pepper in a few more to make things a bit more legible (in my HUMBLE OPINION, take or leave):

```
def get_float_people_id(userdata):
    if userdata.float_people_id:
        return userdata.float_people_id

    float_id = [
        p['people_id'] for p in get_float_data(endpoint='people').json()['people']
        if p['im'] == userdata.user.username
    ]

    if not float_id:
        return None

    # Use most recent Float people_id in the rare case a
    # person has multiple people_ids.
    userdata.float_people_id = int(float_id[-1])
    return userdata.float_people_id
```